test/e2e/proxy_test.go

```diff
@@ -284,16 +282,10 @@ func testLoadBalancerLocalFromNode(t *testing.T, data *TestData, nodes, healthUr
 	}
 }
 
-func testLoadBalancerLocalFromPod(t *testing.T, data *TestData, pods []string, url string, expectedClientIPs, expectedHostnames []string) {
+func testLoadBalancerLocalFromPod(t *testing.T, data *TestData, pods []string, url string) {
 	errMsg := "Service NodePort whose externalTrafficPolicy is Local should be able to be connected from Pod"
-	for idx, pod := range pods {
-		hostname, err := probeHostnameFromPod(data, pod, busyboxContainerName, url)
-		require.NoError(t, err, errMsg)
-		require.Equal(t, hostname, expectedHostnames[idx])
-
-		clientIP, err := probeClientIPFromPod(data, pod, busyboxContainerName, url)
-		require.NoError(t, err, errMsg)
-		require.Equal(t, clientIP, expectedClientIPs[idx])
```

Comment:
Is the clientIP expected to change?

Reply:
For non-host-network Endpoint, the clientIP should be always the same (source Pod IP), but for host-network Endpoint, if the Endpoint is on local host-network, the clientIP should be the source Pod IP; if the Endpoint is on a remote Node, the clientIP will be the transparent interface IP. In this test, we have two host-network Endpoint, and both of them can be selected even when ExternalTrafficPolicy is Local for client from Pod.

Reply:
what's transparent interface IP?

Reply:
It should be Node IP. For example, if the Endpoint is a host network on another Node, the packet will be forwarded to local Node host network, then will be SNATed to remote Node. The SNATed IP should be the local Node IP.

Reply:
OK, it seems the SNAT is not necessary but I understand it's not related to the PR.

Reply:
For example, assuming that there are two Nodes:
Node A: 192.168.77.100, Pod CIDR 10.10.0.0/24
Node B: 192.168.77.101, Pod CIDR 10.10.1.0/24
Service: 10.96.0.9:8080, two Endpoints, 192.168.77.100:8080, 192.168.77.101:8080
Pod on Node A: 10.10.0.10

Previously, if externalTrafficPolicy is Local, for Pod -> Service, only Endpoint 192.168.77.100:8080 will be selected, and the clientIP is 10.10.0.10. Now, Endpoint 192.168.77.101:8080 could be also selected. If so:

- on Pod network: 10.10.0.10 -> 10.96.0.9
- on host network: 10.10.0.10 -> 192.168.77.101
- to make sure that the reply packets can be returned correctly, on network between Nodes: 192.168.77.100 -> 192.168.77.101, then the clientIP is 192.168.77.100. As a result, the clientIP is not certain. It could be 10.10.0.10 or 192.168.77.100.